### ISSUE TITLE
Fluent Checkbox doesn't support aria-checked with Narrator when aria role="treeitem"

### ISSUE THREAD

**Issue by MightySameerDesai** [microsoft/fluentui#16983]
### Environment Information

- **Package version(s)**: "@fluentui/react": "^7.160.0"
- **Browser and OS versions**: Edge 88.0.705.68 (Official build) (64-bit), Windows 10 Enterprise

### Describe the issue:
I have a complex UI component where I have a nested list of items. Each item is a checkbox item that can expand/collapse. So when focus moves to the checkbox, I want the narrator to read out the expanded state, checked state and position in list. 

For this, I set the aria-role to "treeitem" in case of FluentCheckbox by overriding the html props. This however doesn't read out the checked state when in fact it should as per the WAI-ARIA 1.1 documentation where the checked state is supported on this role.

### Please provide a reproduction of the issue in a codepen:
https://codepen.io/SameerDesai2020/pen/ZEBLKOa

#### Actual behavior:
 Reads the list item position level, and expanded state

#### Expected behavior:

Must read, all of 
a. List item position
b. Expand/Collapse state
c. Checked state

### Documentation describing expected behavior

https://www.w3.org/TR/wai-aria-1.1/#treeitem

**Comment by smhigley** [comment]
@MightySameerDesai treeitems use a selected state rather than a checked one, so it would need to have `aria-selected` rather than `aria-checked`. For that reason, the Checkbox control is not a good base for creating a tree.

Unfortunately, we do not currently have tree control, but it's been discussed and would certainly be a good feature request!

**Comment by smhigley** [comment]
Just a small side note -- it would not be a good idea to manually put `aria-selected` on the checkbox, since checkboxes already have native checked semantics. This could cause conflicts and unintended dual checked/selected state, which some screen readers may actually announce. Additionally, an `<input type="checkbox">` does not support having `role="treeitem"`: https://w3c.github.io/html-aria/#el-input-checkbox

In short, Checkbox is best left out of trees 😄.

I'm closing this issue, but please feel welcome to open another issue for a tree request!
